src/GovUk.Education.ExploreEducationStatistics.Content.Search.FunctionApp/Services/CheckSearchableDocuments/ReleaseSummaryRetriever.cs

```diff
@@ -0,0 +1,35 @@
+﻿using GovUk.Education.ExploreEducationStatistics.Content.Search.FunctionApp.Clients.ContentApi;
+using GovUk.Education.ExploreEducationStatistics.Content.Search.FunctionApp.Domain;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Search.FunctionApp.Services.CheckSearchableDocuments;
+
+public interface IReleaseSummaryRetriever
+{
+    Task<IList<ReleaseSummary>> GetAllPublishedReleaseSummaries(
+        CancellationToken cancellationToken = default);
+}
+
+public class ReleaseSummaryRetriever(Func<IContentApiClient> contentApiClientFactory) : IReleaseSummaryRetriever
+{
+    public async Task<IList<ReleaseSummary>> GetAllPublishedReleaseSummaries(
+        CancellationToken cancellationToken = default)
+    {
+        // Get a list of all publication releases
+        var contentApiClient = contentApiClientFactory();
+        var allLivePublicationInfos = await contentApiClient.GetAllLivePublicationInfos(cancellationToken);
```

Comment:
Just looked further into this and I think it's limited to getting 30 publications from the first page of results. Isn't that going to be an issue?